cpp/util/etcd.cc

```diff
@@ -688,14 +726,24 @@ void EtcdClient::FetchDone(RequestState* etcd_req, Task* task) {
   VLOG(2) << "EtcdClient::FetchDone: " << task->status();
 
   if (!task->status().ok()) {
-    // TODO(pphaneuf): If there is a connection problem, we should re-do the
-    // request on another node.
+    if (task->status().error_code() == util::error::UNAVAILABLE) {
+      // Seems etcd wasn't available; pick a new etcd server and retry
+      LOG(WARNING) << "Etcd fetch failed: " << task->status() << ", retrying "
+                   << "on next etcd server.";
+      ChooseNextServer();
+      etcd_req->SetHostPort(endpoint_);
```

Comment:
Accessing `endpoint_` without the lock here (assuming we want `endpoint_` to continue existing, which I don't think so at the moment).